src/NzbDrone.Core/DecisionEngine/DownloadDecisionComparer.cs

```diff
@@ -0,0 +1,129 @@
+        public delegate int CompareDelegate<TSubject, TValue>(DownloadDecision x, DownloadDecision y);
+
+        public DownloadDecisionComparer(IDelayProfileService delayProfileService)
+        {
+            _delayProfileService = delayProfileService;
+        }
+
+        public int Compare(DownloadDecision x, DownloadDecision y)
+        {
+            var comparers = new List<CompareDelegate>
+            {
+                CompareQuality,
+                CompareProtocol,
+                CompareEpisodeCount,
+                CompareEpisodeNumber,
+                ComparePeers,
+                CompareAge,
+                CompareSize
+            };
+
+            return comparers.Select(comparer => comparer(x, y)).FirstOrDefault(result => result != 0);
+        }
+
+        private int Compare<TSubject, TValue>(TSubject left, TSubject right, Func<TSubject, TValue> funcValue)
+            where TValue : IComparable<TValue>
+        {
+            var leftValue = funcValue(left);
+            var rightValue = funcValue(right);
+
+            return leftValue.CompareTo(rightValue);
+        }
+
+        private int CompareReverse<TSubject, TValue>(TSubject left, TSubject right, Func<TSubject, TValue> funcValue)
+            where TValue : IComparable<TValue>
+        {
+            return Compare(left, right, funcValue)*-1;
+        }
+
+        private int Compare(params int[] comparers)
+        {
+            return comparers.Select(comparer => comparer).FirstOrDefault(result => result != 0);
+        }
+
+        private int CompareQuality(DownloadDecision x, DownloadDecision y)
+        {
+            return Compare(Compare(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode => remoteEpisode.Series.Profile.Value.Items.FindIndex(v => v.Quality == remoteEpisode.ParsedEpisodeInfo.Quality.Quality)),
+                           Compare(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode => remoteEpisode.ParsedEpisodeInfo.Quality.Revision.Real),
+                           Compare(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode => remoteEpisode.ParsedEpisodeInfo.Quality.Revision.Version));
+        }
+
+        private int CompareProtocol(DownloadDecision x, DownloadDecision y)
+        {
+            var result = Compare(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode =>
+            {
+                var delayProfile = _delayProfileService.BestForTags(remoteEpisode.Series.Tags);
+                var downloadProtocol = remoteEpisode.Release.DownloadProtocol;
+                return downloadProtocol == delayProfile.PreferredProtocol;
+            });
+
+            return result;
+        }
+
+        private int CompareEpisodeCount(DownloadDecision x, DownloadDecision y)
+        {
+            return Compare(Compare(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode => remoteEpisode.ParsedEpisodeInfo.FullSeason),
+                           CompareReverse(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode => remoteEpisode.Episodes.Count));
```

Comment:
In case of a pilot double, this would mean it prefers a single episode over a double. The same rules as season pack applies, it has to try the double first, coz otherwise it may succeed at E01 but fail at E02 and never get it again coz the double is rejected for not being an upgrade foe E01.